### PR TITLE
fix: btc decode address

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

Adds a wrapper function for `btcUtils.DecodeAddress`
Note: This PR only adds a solution to avoid panic. The original reason that can create these malformed addresses might be a more significant concern and is being looked into separately.

Closes: (https://github.com/zeta-chain/node/issues/1527)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: common/address.go [hunk 1]
```diff
@@ -41,3 +44,45 @@ func (addr Address) IsEmpty() bool {
...
+		ok := IsValidPrefix(prefix, chainId)
+		if !ok {
+			return nil, fmt.Errorf("invalid prefix:%s,chain-id:%d", prefix, chainId)
+		}
+		addressString := inputAddress[oneIndex+1:]
+		if len(addressString) != 39 {
+			return nil, fmt.Errorf("invalid address length:%d,inputaddress:%s", len(addressString), inputAddress)
+		}
+	}
+	address, err = btcutil.DecodeAddress(inputAddress, chainParams)
```
Comment (brewmaster012) [inline common/address.go:86]: for good measure, maybe add a check here: 
`address.IsForNet(chainParams)`
to check that the address is indeed for the intended network

Comment (kingpinXD) [reply common/address.go:74]: cd7b16339aba1fd1623ec62346150517f9a201d2

File: common/address.go [hunk 2]
```diff
@@ -41,3 +44,45 @@ func (addr Address) IsEmpty() bool {
...
+		}
+	}()
+	chainParams, err := GetBTCChainParams(chainId)
+	if err != nil {
+		return nil, err
+	}
+	if chainParams == nil {
+		return nil, fmt.Errorf("chain params not found")
+	}
+	oneIndex := strings.LastIndexByte(inputAddress, '1')
```
[the remark this answers is not in the document]
Comment (kingpinXD) [reply common/address.go]: removed prefix check based on our discussion

File: common/address.go [hunk 3]
```diff
@@ -41,3 +44,37 @@ func (addr Address) IsEmpty() bool {
...
+	case string:
+		return errors.New(x)
+	case error:
+		return x
+	default:
+		return errors.New(fmt.Sprint(x))
+	}
+}
+
+func DecodeBtcAddress(inputAddress string, chainId int64) (address btcutil.Address, err error) {
```
Comment (lumtis) [inline common/address.go:59]:
```suggestion
func DecodeBtcAddress(inputAddress string, chainID int64) (address btcutil.Address, err error) {
```

File: common/address.go [hunk 4]
```diff
@@ -41,3 +44,37 @@ func (addr Address) IsEmpty() bool {
...
 }
+
+func ConvertRecoverToError(r interface{}) error {
+	switch x := r.(type) {
+	case string:
+		return errors.New(x)
+	case error:
+		return x
+	default:
+		return errors.New(fmt.Sprint(x))
```
Comment (lumtis) [inline common/address.go:55]:
```suggestion
		return errors.New(fmt.Sprint("%v", x))
```

File: common/address.go [hunk 5]
```diff
@@ -41,3 +44,37 @@ func (addr Address) IsEmpty() bool {
...
+	}
+}
+
+func DecodeBtcAddress(inputAddress string, chainId int64) (address btcutil.Address, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = ConvertRecoverToError(r)
+			err = fmt.Errorf("input address:%s,chainId:%d,err:%s", inputAddress, chainId, err.Error())
```
Comment (lumtis) [inline common/address.go:63]:
```suggestion
			err = fmt.Errorf("input address:%s, chainId:%d, err:%s", inputAddress, chainId, err.Error())
```

File: common/address.go [hunk 6]
```diff
@@ -41,3 +44,37 @@ func (addr Address) IsEmpty() bool {
...
 }
+
+func ConvertRecoverToError(r interface{}) error {
+	switch x := r.(type) {
+	case string:
+		return errors.New(x)
+	case error:
+		return x
+	default:
+		return errors.New(fmt.Sprint("%v", x))
```
Comment (lumtis) [inline common/address.go:55]:
```suggestion
		return fmt.Errorf("%v", x)
```
Actually better
Might also be related to the lint error

**Review by lumtis — APPROVED** [review]

**Review by brewmaster012 — APPROVED** [review]
LGTM

**Pull request merged**